File: app/src/main/java/hm/orz/chaos114/android/tumekyouen/network/GsonAutoValueAdapterFactory.java
package hm.orz.chaos114.android.tumekyouen.network;

import com.google.gson.TypeAdapterFactory;
import com.ryanharter.auto.value.gson.GsonTypeAdapterFactory;

@GsonTypeAdapterFactory
public abstract class GsonAutoValueAdapterFactory implements TypeAdapterFactory {
    public static TypeAdapterFactory create() {
        return new AutoValueGson_GsonAutoValueAdapterFactory();
    }
}

File: app/src/main/java/hm/orz/chaos114/android/tumekyouen/network/GsonAutoValueAdapterFactory.kt
package hm.orz.chaos114.android.tumekyouen.network

import com.google.gson.TypeAdapterFactory
import com.ryanharter.auto.value.gson.GsonTypeAdapterFactory

@GsonTypeAdapterFactory
abstract class GsonAutoValueAdapterFactory : TypeAdapterFactory {
    companion object {
        fun create(): TypeAdapterFactory {
            return AutoValueGson_GsonAutoValueAdapterFactory()
        }
    }
}
